packages/nimble-components/src/select/tests/select.spec.ts

```diff
@@ -831,413 +843,462 @@ describe('Select', () => {
+                await waitForUpdatesAsync();
+                element.appendChild(option);
+                await waitForUpdatesAsync();
 
-            expect(pageObject.getSelectedOption()?.value).toBe('one');
-        });
+                expect(option.visuallyHidden).toBeTrue();
+            });
 
-        it('dismissing dropdown by clicking away after navigation and then filtering to no options, does not update value', async () => {
-            await clickAndWaitForOpen(element);
-            pageObject.pressArrowDownKey();
-            pageObject.pressCharacterKey('?');
-            await pageObject.clickAway();
+            it('option that is not hidden but visuallyHidden, if removed from DOM, and then re-added to DOM while dropdown closed, is visible in dropdown when opened', async () => {
+                const option = element.options[1] as ListOption;
+                await pageObject.openAndSetFilterText('Three');
+                expect(option.visuallyHidden).toBeTrue();
 
-            expect(pageObject.getSelectedOption()?.value).toBe('one');
-        });
+                element.removeChild(option);
+                await waitForUpdatesAsync();
+                await pageObject.clickAway(); // clears filter
+                element.appendChild(option);
+                await waitForUpdatesAsync();
+                pageObject.clickSelect();
 
-        it('option that is not hidden but visuallyHidden, if removed from DOM, and then re-added to DOM while dropdown open, is not displayed in dropdown', async () => {
-            const option = element.options[1] as ListOption;
-            await pageObject.openAndSetFilterText('Three');
-            expect(option.visuallyHidden).toBeTrue();
+                expect(option.visuallyHidden).toBeFalse();
+            });
 
-            element.removeChild(option);
-            await waitForUpdatesAsync();
-            element.appendChild(option);
-            await waitForUpdatesAsync();
+            it('option that is not hidden but is visuallyHidden, if removed and then added back, will be visible', async () => {
+                const option = element.options[0]! as ListOption;
+                option.visuallyHidden = true;
+                element.removeChild(option);
+                await waitForUpdatesAsync();
+                element.appendChild(option);
+                await waitForUpdatesAsync();
 
-            expect(option.visuallyHidden).toBeTrue();
+                expect(option.visuallyHidden).toBeFalse();
+            });
         });
 
-        it('option that is not hidden but visuallyHidden, if removed from DOM, and then re-added to DOM while dropdown closed, is visible in dropdown when opened', async () => {
-            const option = element.options[1] as ListOption;
-            await pageObject.openAndSetFilterText('Three');
-            expect(option.visuallyHidden).toBeTrue();
+        describe('with filterMode = manual', () => {
```

Comment:
Is there anything interesting to test about the `filterMode` being changed?

Reply:
I don't believe so. There's no lingering state to be concerned about in either mode.